api/engineapi/contracts_actions.py

```diff
@@ -0,0 +1,633 @@
+
+    if offset is not None:
+        query = query.offset(offset)
+
+    query = query.limit(limit)
+
+    return query.all()
+
+
+def delete_registered_contract(
+    db_session: Session,
+    moonstream_user_id: uuid.UUID,
+    registered_contract_id: uuid.UUID,
+) -> RegisteredContract:
+    """
+    Delete a registered contract
+    """
+    try:
+        registered_contract = (
+            db_session.query(RegisteredContract)
+            .filter(RegisteredContract.moonstream_user_id == moonstream_user_id)
+            .filter(RegisteredContract.id == registered_contract_id)
+            .one()
+        )
+
+        db_session.delete(registered_contract)
+        db_session.commit()
+    except Exception as err:
+        db_session.rollback()
+        logger.error(repr(err))
+        raise
+
+    return registered_contract
+
+
+def request_calls(
+    db_session: Session,
+    moonstream_user_id: uuid.UUID,
+    registered_contract_id: uuid.UUID,
+    call_specs: List[data.CallSpecification],
+    ttl_days: Optional[int] = None,
+) -> None:
+    """
+    Batch creates call requests for the given registered contract.
+    """
+    # Check that ttl_days is positive (if specified)
+    if ttl_days is not None and ttl_days <= 0:
+        raise ValueError("ttl_days must be positive")
+
+    # Check that the moonstream_user_id matches the RegisteredContract
+    try:
+        registered_contract = (
+            db_session.query(RegisteredContract)
+            .filter(
+                RegisteredContract.id == registered_contract_id,
+                RegisteredContract.moonstream_user_id == moonstream_user_id,
+            )
+            .one()
+        )
+    except NoResultFound:
+        raise ValueError("Invalid registered_contract_id or moonstream_user_id")
+
+    # Normalize the caller argument using Web3.toChecksumAddress
+    contract_type = ContractType(registered_contract.contract_type)
+    for specification in call_specs:
+        normalized_caller = Web3.toChecksumAddress(specification.caller)
+
+        # Validate the method and parameters for the contract_type
+        validate_method_and_params(
+            contract_type, specification.method, specification.parameters
+        )
+
+        # Calculate the expiration time (if ttl_days is specified)
+        expires_at_sql = None
+        if ttl_days is not None:
+            expires_at_sql = text(f"(NOW() + INTERVAL '{ttl_days} days')")
+
+        request = CallRequest(
+            registered_contract_id=registered_contract.id,
+            caller=normalized_caller,
+            moonstream_user_id=moonstream_user_id,
+            method=specification.method,
+            parameters=specification.parameters,
+            expires_at=expires_at_sql,
+        )
+
+        db_session.add(request)
+    # Insert the new rows into the database in a single transaction
+    try:
+        db_session.commit()
+    except Exception as e:
```

Comment:
Maybe good idea return or number of registered calls or list of `request` object.